viite-UI/src/view/RoadAddressBrowserWindow.js

```diff
@@ -154,75 +170,119 @@
         }
 
         function showData(results, table) {
-            $('#exportAsExcelFile').prop("disabled", false);
-            if (results.length <= MAX_ROWS_TO_DISPLAY) {
+            if (results.length === 0) {
+                roadAddrBrowserWindow.append($('<p id="tableNotification"><b>Hakuehdoilla ei löytynyt yhtäkään osumaa</b></p>'));
+                roadAddrBrowserWindow.append(table.hide());
+            }
+            else if (results.length <= MAX_ROWS_TO_DISPLAY) {
                 roadAddrBrowserWindow.append(table);
-
-            } else {
+                $('#exportAsExcelFile').prop("disabled", false); // enable Excel download button
+            }
+            else {
                 // hide the results and notify user to download result table as excel file
-                roadAddrBrowserWindow.append($('<p id="tableTooBigNotification"><b>Tulostaulu liian suuri, lataa tulokset excel taulukkona</b></p>'));
+                roadAddrBrowserWindow.append($('<p id="tableNotification"><b>Tulostaulu liian suuri, lataa tulokset excel taulukkona</b></p>'));
                 roadAddrBrowserWindow.append(table.hide());
+                $('#exportAsExcelFile').prop("disabled", false); // enable Excel download button
             }
         }
 
 
         function toggle() {
             $('.container').append('<div class="modal-overlay confirm-modal"><div class="modal-dialog"></div></div>');
             $('.modal-dialog').append(roadAddrBrowserWindow.toggle());
+            addDatePicker();
             bindEvents();
         }
 
+        function addDatePicker() {
+            const dateInput = $('#roadAddrStartDate');
+            datePicker = dateutil.addSingleDatePicker(dateInput);
+        }
+
+        function destroyDatePicker() {
+            datePicker.destroy();
+        }
+
         function getCurrentDate() {
             const today = new Date();
             const dayInNumber = today.getDate();
             const day = dayInNumber < 10 ? '0' + dayInNumber.toString() : dayInNumber.toString();
             const monthInNumber = today.getMonth() + 1;
             const month = monthInNumber < 10 ? '0' + monthInNumber.toString() : monthInNumber.toString();
             const year = today.getFullYear().toString();
+            return day + '.' + month + '.' + year;
+        }
+
+        // converts date object to string "yyyy/mm/dd"
```

Comment:
Vai
// converts date object to string "yyyy-mm-dd"?

(eli viivat kauttaviivan paikoilla?)